base/sparse/sparsematrix.jl

```diff
@@ -2569,7 +2569,7 @@ function setindex!(A::SparseMatrixCSC{Tv,Ti}, B::SparseMatrixCSC{Tv,Ti}, I::Abst
     elseif !issortedI
         pI = sortperm(I); @inbounds I = I[pI]
         B = B[pI,:]
-    else !issortedJ
+    elseif !issortedJ
```

Comment:
Could you think of a test which would catch this typo-error?

Reply:
If some auto-formatting tools are used when writting code, this type of error is easy to identify when typing, because of indentation.
Or some linter can detect this pattern and issue a warning.